src/tools/rbd_mirror/main.cc

```diff
@@ -58,14 +58,18 @@ int main(int argc, const char **argv)
   register_async_signal_handler_oneshot(SIGINT, handle_signal);
   register_async_signal_handler_oneshot(SIGTERM, handle_signal);
 
-  mirror = new rbd::mirror::Mirror(g_ceph_context);
+  rbd::mirror::Mirror *mirror =
+    new rbd::mirror::Mirror(g_ceph_context);
   int r = mirror->init();
   if (r < 0) {
     std::cerr << "failed to initialize: " << cpp_strerror(r) << std::endl;
     goto cleanup;
   }
 
-  mirror->run();
+  while (!stopping.read()) {
+    struct timeval sleep_time{1, 0};
+    ::select(0, NULL, NULL, NULL, &sleep_time);
```

Comment:
I used select() here because I knew it would interrupt if signal were received (so no delay before shutdown). I will check if sleep()/sleep_for.